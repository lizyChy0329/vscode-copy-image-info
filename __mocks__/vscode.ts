import { createVSCodeMock } from 'jest-mock-vscode'
import { vi } from 'vitest'

export const vscode = createVSCodeMock(vi)
