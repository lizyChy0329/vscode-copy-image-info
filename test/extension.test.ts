import path from 'node:path'
import { describe, expect, it, vi } from 'vitest'
import * as vscode from 'vscode'
import { URI } from 'vscode-uri'

vi.mock('vscode')

export function sum(a: number, b: number) {
  return a + b
}

const testImagePath = path.resolve(__dirname, 'src/assets/103976180.jpg')
// 假设这是您要测试的命令
const commandId = 'copy-image-size.copyImageExt'

describe('vscode-copy-image-size Command Tests', () => {
  it('should execute the command successfully', async () => {
    const result = await vscode.commands.executeCommand(commandId, URI.file(testImagePath))
    expect(result).toMatchSnapshot()
  })

  it('adds 1 + 2 to equal 3', () => {
    expect(sum(1, 2)).toBe(3)
  })
})
