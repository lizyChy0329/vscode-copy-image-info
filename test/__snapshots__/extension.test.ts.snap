// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vscode-copy-image-size Command Tests > should execute the command successfully 1`] = `undefined`;
